kiteco/KiteSublime
# Support JavaScript Beta

Resolves https://github.com/kiteco/kiteco/issues/10591
Addresses part of https://github.com/kiteco/kiteco/issues/10456

### What was done

Turned on JavaScript beta, added beta notifications per language, and changed onboarding endpoint call to use language specific params (per the lexical spec).

<img width="508" alt="Screen Shot 2020-03-30 at 12 49 31 PM" src="https://user-images.githubusercontent.com/18232816/77955792-aacd0080-7285-11ea-842f-c4d35eaae1a4.png">
<img width="524" alt="Screen Shot 2020-03-30 at 1 12 42 PM" src="https://user-images.githubusercontent.com/18232816/77957319-2d56bf80-7288-11ea-9741-93e3b26abb44.png">

### Considerations

See discussion in PR comments below for changes. Essentially we'll try showing an onboarding file and fallback to a beta notification for lexical for the time being. Python falls back to a specific help dialog linking at the moment to this repo.

~As of 2020/03/30, we do not yet have onboarding files for sublime JavaScript or Go so onboarding for those languages defaults to this popup:~

<img width="592" alt="Screen Shot 2020-03-30 at 12 58 56 PM" src="https://user-images.githubusercontent.com/18232816/77956133-3ba3dc00-7286-11ea-87f8-9bdd41cd5cd9.png">

There's no support in the Sublime API for less intrusive notifications at the moment.

### QA

Check for JavaScript completions
Check that with JS or Go disabled and all user settings of form `show_<lang>_help` set to true that the onboarding notification / file does not open.
Check that with JS and Go enabled and all user settings of form `show_<lang>_help` set to true:
1. onboarding notification / file opens for lexical languages once (e.g. after seeing either an onboarding file or a notification for Go, JS does not show onboarding but Python does.)
2. Seeing onboarding for Python does not subsequently show lexical onboarding when a lexical supported file is open.
